### PR TITLE
aggregate child clusters workload status to desc

### PR DESCRIPTION
**Pull request by aven-ai**
#### What type of PR is this?
Kind/feature

#### What this PR does / why we need it:
aggregate child clusters workload status to desc

#### Which issue(s) this PR fixes:

Fixes

### CODE REVIEW

**Comment by aven-ai** [comment]
@dixudx @lmxia please help to review this, thanks.

**Comment by aven-ai** [comment]
@dixudx @lmxia please help to review this, thanks.

File: pkg/agent/deployer/generic/generic.go [hunk 1]
```diff
@@ -226,3 +228,100 @@ func (deployer *Deployer) AddController(gvk schema.GroupVersionKind, controller
...
+	var err error
+	if !reflect.DeepEqual(desc.Status.ManifestStatuses, descStatus.ManifestStatuses) {
+		err = deployer.UpdateDescriptionStatus(desc, descStatus)
+		klog.V(5).Infof("SyncDescriptionStatus Descriptions manifestStatus has changed, UpdateStatus. err: %s", err)
+	}
+
+	return err
+}
+
+func (deployer *Deployer) UpdateDescriptionStatus(desc *appsapi.Description, status *appsapi.DescriptionStatus) error {
```
Comment (lmxia) [inline pkg/agent/deployer/generic/generic.go:304]: Duplicate defined.  Use utils.UpdateDescriptionStatus(desc, descStatus, deployer.clusternetClient)

File: pkg/controllers/apps/resource/resource.go [hunk 2]
```diff
@@ -69,25 +102,43 @@ func NewController(apiResource *metav1.APIResource, clusternetClient *versioned.
...
+			newresource := newObj.(*unstructured.Unstructured)
+			oldstatus, _, _ := unstructured.NestedMap(oldresource.Object, "status")
+			newstatus, _, _ := unstructured.NestedMap(newresource.Object, "status")
+			oldspec, _, _ := unstructured.NestedMap(oldresource.Object, "spec")
+			newspec, _, _ := unstructured.NestedMap(newresource.Object, "spec")
+			val, ok := oldresource.GetAnnotations()[known.ObjectOwnedByDescriptionAnnotation]
+
+			var resAttrs *ResourceAttrs
+			var err error
+			if !reflect.DeepEqual(oldspec, newspec) {
```
Comment (lmxia) [inline pkg/controllers/apps/resource/resource.go:122]:
```
if !reflect.DeepEqual(oldstatus, newstatus) {
     resAttrs, err = getResourceAttrs(val, ObjectUpdateStatus)
} else {
      resAttrs, err = getResourceAttrs(val, ObjectUpdateOthers)
}

File: pkg/utils/deployer.go [hunk 3]
```diff
@@ -385,9 +386,15 @@ func ApplyDescription(ctx context.Context, clusternetClient *clusternetclientset
 	}
 
 	// update status
-	desc.Status.Phase = statusPhase
-	desc.Status.Reason = reason
-	_, err := clusternetClient.AppsV1alpha1().Descriptions(desc.Namespace).UpdateStatus(context.TODO(), desc, metav1.UpdateOptions{})
+	Status := desc.Status.DeepCopy()
```
Comment (lmxia) [inline pkg/utils/deployer.go:389]: Local variable "Status" no need start with upcase.

File: pkg/agent/deployer/generic/generic.go (hunk 1 above)
Comment (aven-ai) [reply pkg/agent/deployer/generic/generic.go]: > Duplicate defined. Use utils.UpdateDescriptionStatus(desc, descStatus, deployer.clusternetClient)

done.

File: pkg/controllers/apps/resource/resource.go (hunk 2 above)
Comment (aven-ai) [reply pkg/controllers/apps/resource/resource.go:122]: > ```
> if !reflect.DeepEqual(oldstatus, newstatus) {
>      resAttrs, err = getResourceAttrs(val, ObjectUpdateStatus)
> } else {
>       resAttrs, err = getResourceAttrs(val, ObjectUpdateOthers)
> }
> ```
If spec and status change at the same time, the logic here needs to process spec first.

**Review by dixudx — APPROVED** [review]

**Pull request merged**